src/web-animations-next-animation.js

```diff
@@ -25,13 +27,48 @@
     this._animation = null;
     this._childAnimations = [];
     this._callback = null;
+    this._oldPlayState = 'idle';
     this._rebuildUnderlyingAnimation();
     // Animations are constructed in the idle state.
     this._animation.cancel();
+    this._updatePromises();
   };
 
   scope.Animation.prototype = {
+    _updatePromises: function() {
+      var oldPlayState = this._oldPlayState;
+      var newPlayState = this.playState;
+      if (this._readyPromise && newPlayState !== oldPlayState) {
+        if (newPlayState == 'idle') {
+          if (this._readyPromiseState == 'pending') {
```

Comment:
> You still use them to filter down your list of animations with promises. This is why I only suggested you could simplify it, I didn't have the whole story :-)

Oh yeah. I was only checking in web-animations-next-animation. We do still need it. OK, sure, I guess we can remove this check.